jwst/spectral_leak/spectral_leak_step.py

```diff
@@ -0,0 +1,100 @@
+#! /usr/bin/env python
+
+from stdatamodels.jwst import datamodels
+from jwst.datamodels import ModelContainer
+from ..stpipe import Step
+import numpy as np
+
+from . import spectral_leak
+
+__all__ = ["SpectralLeakStep"]
+
+
+class SpectralLeakStep(Step):
+    """
+    The MIRI MRS has a spectral leak in which 6 micron light leaks into the
+    12 micron channel.  This step applies a correction to the 12 micron channel.
+    """
+
+    class_alias = "spectral_leak"
+
+    reference_file_types = ['mrsptcorr']
+
+    def process(self, input):
+        """Execute the step.
+
+        Parameters
+        ----------
+        input : model container
+            List of input 1D spectral data models
+
+        Returns
+        -------
+        result : data model
+            The corrected data model. This will be "input" if ...
+            This will be `input` if the step was skipped; otherwise,
+            it will be a corrected 1-D extracted spectra that contains
+            the  3B MRS range.
+        """
+
+        ch1b = None
+        ch3a = None
+        ich3a = None
+        ch1b_wave = 6.0
+        ch3a_wave = 12.0
+
+        with datamodels.open(input) as input_model:
+            if isinstance(input_model, ModelContainer):
+                # Retrieve the reference parameters for this type of data
+                sp_leak_ref = self.get_reference_file(input_model, 'mrsptcorr')
+
+                for i, x1d in enumerate(input_model):  # input_model is a Model Container
+                    # check that we have the correct type of data
+                    if not isinstance(x1d, datamodels.MultiSpecModel):
+                        self.log.warning("Data sent to spectral_leak step is not an extracted spectrum. It is  {}." .format(type(x1d)))
+                        return input
+
+                    channel = x1d.meta.instrument.channel
+                    band = x1d.meta.instrument.band
+                    srctype = x1d.spec[0].source_type
+                    if srctype == 'EXTENDED':
+                        self.log.info('No spectral leak correction for extended source data')
```

Comment:
I think this should be a warning message, not info, and the warning should also say that the step is being skipped. Then set the completion status to SKIPPED before returning `input`.  Same for the warning in line 54 (should say the step is being skipped, set the status to SKIPPED, and then return `input`).

Reply:
Ok I have a question for you. 'input' will be a DataModelContainter.
So before I return and set the step to skip do I just do this: (I will try and make this fail and see what I need)
Just wondered if you knew off the top of your head

self.log.warning("Data sent to spectral_leak step is not an extracted spectrum. It is  {}." .format(type(x1d)))
input .meta.cal_step.spectral_leak = 'COMPLETE'.  # IS THIS VALID FOR THIS TYPE OF DATA
return input